Source/RestKit/JSONValue.swift

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright IBM Corporation 2016-2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+import Foundation
+
+/// A JSON value (one of string, number, object, array, true, false, or null).
+public enum JSONValue: Equatable, Codable {
+    case null                        /// A null value.
+    case boolean(Bool)               /// A boolean value.
+    case string(String)              /// A string value.
+    case int(Int)                    /// A number value, represented as an integer.
+    case double(Double)              /// A number value, represented as a double.
+    case array([JSONValue])          /// An array value.
+    case object([String: JSONValue]) /// An object value.
```

Comment:
This is the only part that really bugs me (and I'm admittedly pedantic about such things).  If something is named `object`, I expect it to be an object, i.e. a `class` instance.  I'd name this `dictionary` - I think that would be clearer and less confusing. (The bad news is that it's used a lot - the good news is that search/replace is your friend).

Reply:
That's a great point. I could go either way, although I lean a bit towards keeping `object` for consistency with the [JSON spec](http://www.json.org/).